vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -250,4 +283,43 @@ public void removeCategory(IVSImage image, ImageCategory category) {
         image.getCategories().remove(category);
         imageRepo.save((VSImage) image);
     }
+
+    /**
+     * This method is used to search the search string specified in the input
+     * 
+     * @param searchTerm - This is the search string which is being searched.
+     * @return list of images based on the search string.
+     */
+    @Override
+    public List < IVSImage > search(String searchTerm) {
+        String likeSearchTerm = "%" + searchTerm + "%";
+        List < VSImage > results = imageRepo.findByFilenameLikeOrNameLikeOrDescriptionLike(likeSearchTerm, likeSearchTerm,
+            likeSearchTerm);
+        List < IVSImage > imageResults = new ArrayList < > ();
+        results.forEach(r - > imageResults.add(r));
+        return imageResults;
+    }
+
+    /**
+     * Method to return the requested images
+     * 
+     * @param pageNo.    if pageNo<1, 1st page is returned, if pageNo>total
+     *                   pages,last page is returned
+     * @param searchTerm - This is the search string which is being searched.
+     * @param images     - This is the result list of search images.
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List < IVSImage > getImagesForPagination(int pageNo, List < IVSImage > images, String searchTerm) {
+        pageNo = validatePageNumber(pageNo, images);
+        Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize);
+        String likeSearchTerm = "%" + searchTerm + "%";
+        Page < VSImage > result = imageRepo.findByFilenameLikeOrNameLikeOrDescriptionLike(sortByRequestedField,
+            likeSearchTerm, likeSearchTerm, likeSearchTerm);
+        List < IVSImage > imageResults = new ArrayList < > ();
+        if (result != null) {
+            result.getContent().forEach(i - > imageResults.add(i));
+        }
+        return imageResults;
```

Comment:
I'm very confused by what this method does. Why does it get a list of images and then also search the repo? It should jus search and return the request page of search results.

Reply:
Requirement says "The search should search filename, name, and description for the given search term"
so in this function i am sending IVS images to validate the page number. and searching in repo to get the list of images for that particular page based on search conditions.

Reply:
but why does it need a list off images for that?

Reply:
I am using an existing function validatePageNumber() which is present in Imageservice.java file, that function has list of images as parameter, so sending the list of images to validate the function which will return page no, then based on the page number requesting images in the specific sorted order.